File: scratch/find_aarohi.py
import psycopg2

def main():
    try:
        conn = psycopg2.connect(
            host="aws-1-ap-south-1.pooler.supabase.com",
            port=5432,
            dbname="postgres",
            user="postgres.uodmjwjnhinbbvexbyvd",
            password="akshitsha84",
            sslmode="require"
        )
        cur = conn.cursor()
        cur.execute("""
            SELECT s.id, s."admissionNumber", u.id, u.email, u."firstName", u."lastName"
            FROM public."Student" s
            JOIN public."User" u ON s."userId" = u.id
            WHERE u."firstName" ILIKE '%Aarohi%' OR u."lastName" ILIKE '%Mishra%' OR u."firstName" ILIKE '%Mishra%' OR u."lastName" ILIKE '%Aarohi%';
        """)
        rows = cur.fetchall()
        print(f"Found {len(rows)} matching students:")
        for r in rows:
            print(f"Student ID: {r[0]} | AdmNo: {r[1]} | UserID: {r[2]} | Email: {r[3]} | Name: {r[4]} {r[5]}")
            
        cur.execute("""
            SELECT id, email, "firstName", "lastName", role
            FROM public."User"
            WHERE "firstName" ILIKE '%Aarohi%' OR "lastName" ILIKE '%Mishra%' OR "firstName" ILIKE '%Mishra%' OR "lastName" ILIKE '%Aarohi%';
        """)
        user_rows = cur.fetchall()
        print(f"\nFound {len(user_rows)} matching users:")
        for ur in user_rows:
            print(f"UserID: {ur[0]} | Email: {ur[1]} | Name: {ur[2]} {ur[3]} | Role: {ur[4]}")
            
        conn.close()
    except Exception as e:
        print("Error:", e)

if __name__ == "__main__":
    main()

File: scratch/find_aarohi.dart
import 'package:supabase/supabase.dart';

void main() async {
  final supabase = SupabaseClient(
    'https://bstevdkjqjzaglayicdg.supabase.co',
    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImJzdGV2ZGtqcWp6YWdsYXlpY2RnIiwicm9sZSI6ImFub24iLCJpYXQiOjE3ODA2MjU5MDUsImV4cCI6MjA5NjIwMTkwNX0.DuFB6mkZLcE2qhhEQITchXjth0h86P6bkQSfY_bbvOE'
  );

  try {
    final users = await supabase
        .from('User')
        .select('id, email, firstName, lastName, role')
        .or('firstName.ilike.%Aarohi%,lastName.ilike.%Mishra%,firstName.ilike.%Mishra%,lastName.ilike.%Aarohi%');

    print('Matching Users: ${users.length}');
    for (var u in users) {
      print(u);
    }

    final students = await supabase
        .from('Student')
        .select('id, admissionNumber, user:User(id, firstName, lastName)')
        .or('user.firstName.ilike.%Aarohi%,user.lastName.ilike.%Mishra%'); // wait, Postgrest doesn't support nested relation or in root, so let's filter after fetching or select users first.
  } catch (e) {
    print('Error: $e');
  } finally {
    supabase.dispose();
  }
}
